mobile/library/cc/engine_builder.cc

```diff
@@ -734,6 +739,15 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     rcv_buf_sock_opt->set_int_value(socket_receive_buffer_size_);
     rcv_buf_sock_opt->set_description(
         absl::StrCat("SO_RCVBUF = ", socket_receive_buffer_size_, " bytes"));
+
+    envoy::config::core::v3::SocketOption* udp_snd_buf_sock_opt =
+        base_cluster->mutable_upstream_bind_config()->add_socket_options();
+    udp_snd_buf_sock_opt->set_name(SO_SNDBUF);
+    udp_snd_buf_sock_opt->set_level(SOL_SOCKET);
+    udp_snd_buf_sock_opt->set_int_value(udp_socket_send_buffer_size_);
+    udp_snd_buf_sock_opt->mutable_type()->mutable_datagram();
```

Comment:
Mind adding a comment 
// Make sure this option is only applied to UDP sockets and not TCP

Reply:
Done.